benchmarks/benchmarks/_utils.py

```diff
@@ -1,29 +1,49 @@
 from __future__ import annotations
 
+import itertools
+import sys
 import warnings
 from functools import cache
 from typing import TYPE_CHECKING
 
 import numpy as np
 import pooch
 from anndata import concat
+from asv_runner.benchmarks.mark import skip_for_params
+from scipy import sparse
 
 import scanpy as sc
 
 if TYPE_CHECKING:
-    from typing import Literal
+    from collections.abc import Callable, Sequence, Set
+    from typing import Literal, Protocol, TypeVar
 
     from anndata import AnnData
 
+    C = TypeVar("C", bound=Callable)
+
+    class ParamSkipper(Protocol):
+        def __call__(self, **skipped: Set) -> Callable[[C], C]: ...
+
     Dataset = Literal["pbmc68k_reduced", "pbmc3k", "bmmc", "lung93k"]
+    KeyX = Literal[None, "off-axis"]
+    KeyCount = Literal["counts", "counts-off-axis"]
 
 
 @cache
 def _pbmc68k_reduced() -> AnnData:
+    """A small datasets with a dense `.X`"""
     adata = sc.datasets.pbmc68k_reduced()
+    adata.var["mt"] = adata.var_names.str.startswith("MT-")
```

Comment:
this seems to be the same everywhere, maybe extract it into `_get_dataset_raw`

Reply:
I don't this this is a bad idea.